pact-request-api/Pact/Core/Command/Client.hs

```diff
@@ -0,0 +1,133 @@
+module Pact.Core.Command.Client where
+
+import Control.Monad
+import Control.Monad.Except
+import Data.ByteString (ByteString)
+import Data.Text (Text)
+import qualified Pact.JSON.Encode as J
+
+import Pact.Core.Capabilities
+import Pact.Core.ChainData
+import Pact.Core.Command.RPC
+import Pact.Core.Command.Types
+import Pact.Core.Command.Util
+import Pact.Core.Command.Crypto
+import Pact.Core.Guards
+import qualified Pact.Core.Hash as PactHash
+import Pact.Core.PactValue
+import Pact.Core.Names
+import Pact.Core.Verifiers
+
+-- CREATING AND SIGNING TRANSACTIONS
+
+type UserCapability = CapToken QualifiedName PactValue
+
+mkCommand
+  :: J.Encode c
+  => J.Encode m
+  => [(Ed25519KeyPair, [UserCapability])]
+  -> [Verifier ParsedVerifierProof]
+  -> m
+  -> Text
+  -> Maybe NetworkId
+  -> PactRPC c
+  -> IO (Command ByteString)
+mkCommand creds vers meta nonce nid rpc = mkCommand' creds encodedPayload
+  where
+    payload = Payload rpc nonce meta (keyPairsToSigners creds) (vers <$ guard (not (null vers))) nid
```

Comment:
I've seen these exact line several times, maybe its worth factoring out.

Reply:
I've pulled out the most confusing repetitive part, `vers <$ guard ...`, into a single documented function.